### PR TITLE
Move

### PR DESCRIPTION
**Pull request by scottcrawfordhtx**
Implement the Move function

### CODE REVIEW

File: AdventureTime/AdventureTime/Player.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            direction = direction.ToLower();
+
+            while (!IsDirectionValid(direction))
+            {
+                direction = Console.ReadLine();
+            }
+            
+                switch (direction)
+                {
+                    case "up":
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/Player.cs:55]: "up" "down" "left" and "right" are magic constants. refactor these into their own file or use an enum.

File: AdventureTime/AdventureTime/Player.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                        break;
+                    case "right":
+                        tempX++;
+                        break;
+                }
+        }
+
+        public bool IsDirectionValid(string direction)
+        {
+            if (direction.ToLower() == "up" || direction.ToLower() == "down" || direction.ToLower() == "left" || direction.ToLower() == "right")
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/Player.cs:72]: same thing here

File: AdventureTime/AdventureTime/GameBoard.cs [hunk 3]
```diff
@@ -10,59 +10,18 @@ public class GameBoard
 
         public GameBoard()
         {
-            Board = GenerateBoard();
+            Board = new int[16, 16];
         }
```
Comment (Ungiebungie) [inline AdventureTime/AdventureTime/GameBoard.cs:15]: I can also write comments, it's pretty cool.

File: AdventureTime/AdventureTime/Player.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            int tempX = X;
+            int tempY = Y;
+
+            Console.WriteLine("What direction would you like to go?");
+
+            string direction = Console.ReadLine();
+            direction = direction.ToLower();
+
+            while (!IsDirectionValid(direction))
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/Player.cs:48]: this was good

File: AdventureTime/AdventureTime/Player.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        public Player()
+        {
+            X = r.Next(10, 26);
+            Y = r.Next(10, 26);
+            HP = 10;
+            Money = 0;
+
+            IsAlive = true;
+
+            if (HP < 1)
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/Player.cs:26]: why are we checking if they're dead when we construct the player

File: AdventureTime/AdventureTime/GameBoard.cs [hunk 6]
```diff
@@ -10,59 +10,18 @@ public class GameBoard
...
 
         public GameBoard(int[,] board)
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/GameBoard.cs:16]: we probably don't need an outside force to generate the game board

File: AdventureTime/AdventureTime/Player.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            }
+        }
+
+        public void Kill()
+        {
+            IsAlive = false;
+            Console.WriteLine("You died");
+        }
+
+        public void Move(GameBoard gameBoard)
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/Player.cs:38]: we're not checking if the player ends up in a valid position after they move

File: AdventureTime/AdventureTime/GameBoard.cs [hunk 8]
```diff
@@ -10,59 +10,18 @@ public class GameBoard
...
-
-        public int EventNum(int x, int y)
-        {
-            return Board[x, y];
-        }
-
         internal bool IsPositionValid(int xCoord, int yCoord)
         {
-            return (Board[xCoord, yCoord] != 0);
+            return (yCoord <= Board.GetLength(0) && xCoord <= Board.GetLength(1));
```
Comment (stu-millner) [inline AdventureTime/AdventureTime/GameBoard.cs:23]: less than *or equal to* will result in an out of bounds exception

File: AdventureTime/AdventureTime/Player.cs (hunk 5 above)
Comment (Ungiebungie) [reply AdventureTime/AdventureTime/Player.cs:26]: Can we talk about benefits? I haven't been getting my paychecks and there doesn't seem to be an HR for this project.

File: AdventureTime/AdventureTime/Program.cs [hunk 9]
```diff
@@ -18,12 +18,12 @@ static void Main(string[] args)
         private static void Intro()
         {
             GameBoard gameboard = new GameBoard();
-            gameboard.PrintBoard();
+            Player player = new Player();
 
             Console.WriteLine("It's so dark in here, and my cock smells of twink-stink. I've gotta get outta here.");
-            Console.ReadKey();
-            Console.WriteLine("Use W,A,S,D to move around the cavern. Good luck.");
```
[the remark this answers is not in the document]
Comment (scottcrawfordhtx) [reply AdventureTime/AdventureTime/Program.cs:25]: Gave the user the direction options

File: AdventureTime/AdventureTime/Player.cs [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        public int X { get; set; }
+        public int Y { get; set; }
+        public bool IsAlive { get; set; }
+        public int HP { get; set; }
+        public int Money { get; set; }
+        Random r = new Random();
+
+        public Player()
+        {
+            X = r.Next(10, 26);
```
[the remark this answers is not in the document]
Comment (scottcrawfordhtx) [reply AdventureTime/AdventureTime/Player.cs:19]: The player now starts at any random position within the cave

File: AdventureTime/AdventureTime/Player.cs (hunk 1 above)
Comment (scottcrawfordhtx) [reply AdventureTime/AdventureTime/Player.cs:55]: Done

File: AdventureTime/AdventureTime/Player.cs (hunk 4 above)
Comment (scottcrawfordhtx) [reply AdventureTime/AdventureTime/Player.cs:48]: Thonks

**Review by stu-millner — APPROVED** [review]

**Pull request merged**